### PR TITLE
daemon: ensure changes are created after error diversions and just before responses

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

**Comment by pedronis** [comment]
requires #1062

File: daemon/api_test.go [hunk 1]
```diff
@@ -100,8 +100,6 @@ func (s *apiSuite) SetUpSuite(c *check.C) {
 func (s *apiSuite) TearDownSuite(c *check.C) {
 	newRemoteRepo = nil
 	muxVars = nil
-	snapstateInstall = snapstate.Install
-	snapstateGet = snapstate.Get
```
Comment (mvo5) [inline daemon/api_test.go]: Good catch! Wonder why this was here instead of in SetUpTest

**Comment by mvo5** [comment]
:+1:

**Comment by chipaca** [comment]
:+1:

**Pull request merged**